cmds/boot/boot.go

```diff
@@ -133,107 +100,171 @@ func getSupportedFilesystem() ([]string, error) {
+// partitioned; and all its partitions.
+func mountEntry(d string, supportedFilesystem []string) error {
 	var err error
-	exist, err := os.Stat("/u-root")
-	if exist == nil {
-		err = syscall.Mkdir("/u-root", 0777)
-		if err != nil {
-			return false, err
-		}
+	verbose("Try to mount %v", d)
+
+	// find or create the mountpoint.
+	m := filepath.Join(uroot, d)
+	if _, err = os.Stat(m); err != nil && os.IsNotExist(err) {
+		err = os.MkdirAll(m, 0777)
 	}
-	var flags uintptr
-	// Was supposed to be unecessary for kernel 4.x.x
-	if verbose {
-		log.Printf("/dev/" + path)
+	if err != nil {
+		verbose("Can't make %v", m)
+		return err
 	}
 	for _, filesystem := range supportedFilesystem {
-		flags = syscall.MS_MGC_VAL
-		// Need to load the filesystem kind supported
-		exist, err = os.Stat("/u-root/" + path)
-		if exist == nil {
-			err = syscall.Mkdir("/u-root/"+path, 0777)
-			if err != nil {
-				return false, err
-			}
-		}
-		err := syscall.Mount(filepath.Join("/dev/",path), filepath.Join("/u-root/",path), filesystem, flags, "")
-		if err == nil {
-			return true, nil
+		// Was supposed to be unecessary for kernel 4.x.x
+		var flags = uintptr(syscall.MS_MGC_VAL | syscall.MS_RDONLY)
+		verbose("\twith %v", filesystem)
+		if err := syscall.Mount(d, m, filesystem, flags, ""); err == nil {
+			return err
 		}
 	}
-	returnValue = false
-	return returnValue, nil
+	verbose("No mount succceeded")
+	return errors.New("Unable to mount any partition on " + d)
 }
 
-func umountEntry(path string) bool {
-	var returnValue bool
-	var flags int
-	// Was supposed to be unecessary for kernel 4.x.x
-	flags = syscall.MNT_DETACH
-	err := syscall.Unmount(path, flags)
-	if err == nil {
-		return true
+func umountEntry(n string) error {
+	return syscall.Unmount(n, syscall.MNT_DETACH)
+}
+
+func loadISOLinux(dir, base string) ([]string, string, string, error) {
+	sol := filepath.Join(dir, base)
+	isolinux, err := ioutil.ReadFile(sol)
+	if err != nil {
+		return nil, "", "", err
 	}
-	returnValue = false
-	return returnValue
+
+	// it's easier we think to do include processing here.
+	lines := strings.Split(string(isolinux), "\n")
+	var result []string
+	for _, l := range lines {
+		f := strings.Fields(l)
+		if len(f) == 0 {
+			continue
+		}
+		switch f[0] {
+		default:
+			result = append(result, l)
+		case "include":
+			i, _, _, err := loadISOLinux(dir, f[1])
+			if err != nil {
+				return nil, "", "", err
+			}
+			result = append(result, i...)
+			// If the string contain include
+			// we must call back as to read the content of the file
+		}
+	}
+	return result, dir, dir, nil
 }
 
 // checkBootEntry is looking for grub.cfg file
 // and return absolute path to it
-func checkBootEntry(mountPoint string) ([]byte, string) {
-	grub, err := ioutil.ReadFile(filepath.Join(mountPoint, "/boot/grub/grub.cfg"))
+func checkBootEntry(mountPoint string) ([]string, string, string, error) {
+	grub, err := ioutil.ReadFile(filepath.Join(mountPoint, "boot/grub/grub.cfg"))
```

Comment:
is this always where the grub config file is?

Reply:
there are no guarantees, we may have to get smarter, but this is pretty standard.

Reply:
RedHat based distros on EFI based systems will install grub.cfg at:

- /boot/efi/EFI/redhat/grub.cfg
- /boot/efi/EFI/fedora/grub.cfg
- /boot/efi/EFI/centos/grub.cfg

There's a symlink at /etc/grub2-efi.cfg, which is the recommended path to use.

Debian based distros on EFI based systems appear to continue to use /boot/grub/grub.cfg.

Reply:
at /etc? What on earth are they thinking? 
For now I'd like to leave this as is, since it's working for our current use case, and address the more complex case with another pr.

Reply:
👍

Reply:
hey ben is that approval or ...